### PR TITLE
Fix panic if syslog is enabled in a container [RHELDST-12206]

### PR DESCRIPTION
**Pull request by dichn**

### CODE REVIEW

File: internal/log/syslog.go [hunk 1]
```diff
@@ -60,6 +61,11 @@ func (h *syslogHandler) HandleLog(e *apexLog.Entry) error {
 	enc := json.NewEncoder(&bld)
 	enc.Encode(syslogFields(e))
 
+	if h.writer == nil {
+		stdlog.Printf("Error logging: Syslog unavailable, please change the logging config")
```
Comment (rohanpm) [inline internal/log/syslog.go:65]: I guess this is going to stdout, can you please make it go to stderr instead?

**Review by rohanpm — CHANGES_REQUESTED** [review]
Minor: about the commit message subject:

`syslog panics in openshift container`

We don't use that style (i.e. summarizing a bug being fixed but not stating that you're fixing it).

The preferred style is to say that you're *fixing* the problem, e.g.

`Fix panic if syslog is enabled in a container`

It can get really confusing if the commit log contains a mixture of messages some like "Fix foo bug" and others just like "foo bug".

**Comment by dichn** [comment]
```
// when syslog is not unavailable
if h.writer == nil {
    stdlog.Printf("Error logging: Syslog unavailable, please change the logging config")
    return nil
}
```
I was using `os.Exit(1)` before, but I found it's quite difficult to make the test coverage 100%[1, 2], then I replaced it with `return nil`, but it did not exit gracefully, but keep calling the `func (i impl) Exec(ctx context.Context, args args.Config) error` in `internal/rsync/rsync.go`.

[1] https://stackoverflow.com/questions/26225513/how-to-test-os-exit-scenarios-in-go
[2] https://stackoverflow.com/questions/40615641/testing-os-exit-scenarios-in-go-with-coverage-information-coveralls-io-goverall

File: internal/log/syslog.go (hunk 1 above)
Comment (dichn) [reply internal/log/syslog.go:65]: Yes, will change to
```
        if h.writer == nil {
-               stdlog.Printf("Error logging: Syslog unavailable, please change the logging config")
+               fmt.Fprintf(os.Stderr, "Error logging: Syslog unavailable, please change the logging config")
                return nil
        }
```

**Comment by dichn** [comment]
Repushed for 

1) return an error when syslog is not available in HandleLog().
2) the commit message subject updated

@rohanpm 
I think this matches apexLog's error handling[1]. And it writes to stderr[2].

[1] https://github.com/apex/log/blob/8da83152b5d6177b4bfe3d12810a5afd25355170/logger.go#L153
[2] https://github.com/golang/go/blob/master/src/log/log.go#L90

**Comment by dichn** [comment]
I think this satisfies the acceptance criteria

(2) continue the rsync without logging, and produce some sort of warning message that the logger will not be used.

So put it ready.

**Review by rohanpm — APPROVED** [review]

**Review by emanuelelevo — APPROVED** [review]

**Pull request merged**